src/Common/src/Interop/Unix/libcrypto/Interop.ERR.cs

```diff
@@ -17,13 +19,56 @@ internal static partial class libcrypto
         [DllImport(Libraries.LibCrypto, CharSet = CharSet.Ansi)]
         private static extern void ERR_error_string_n(uint e, StringBuilder buf, int len);
 
-        internal static string GetOpenSslErrorString()
+        private static string ERR_error_string_n(uint error)
         {
-            uint error = ERR_get_error();
             StringBuilder buf = new StringBuilder(1024);
-            
+
             ERR_error_string_n(error, buf, buf.Capacity);
             return buf.ToString();
         }
+
+        internal static Exception CreateOpenSslCryptographicException()
+        {
+            const uint ReasonMask = 0x0FFF;
+            const uint ERR_R_MALLOC_FAILURE = 64 | 1;
+
+            uint error = ERR_get_error();
+            uint last = error;
+
+            while (last != 0)
+            {
+                last = ERR_get_error();
```

Comment:
Sure, I'll add some comments.  A couple scenarios off-hand are
1. We do some sort of fallback and don't drain the error state out after the first call fails
2. OpenSSL internally ends up calling ERR_set_error twice during one call that we make

Though, it also turns out that by doing more testing here this fix also isn't right... `ERR_get_error`/`ERR_set_error` isn't a stack, but a queue.  So I'll be fixing that, too.  Good thing it's consolidated now :smile:.